src/main/java/com/asem/regestration/model/dao/CourseDAO.java

```diff
@@ -0,0 +1,15 @@
+package com.asem.regestration.model.dao;
+
+import com.asem.regestration.model.entity.Courses;
+import com.asem.regestration.model.entity.Student;
+
+import java.sql.SQLException;
+
+public interface CourseDAO {
+
+    Courses findById(int id) throws SQLException;
```

Comment:
Courses here is plural and confusing as you just return single course but its name is courses

Reply:
Also its very general exception to throw SQLException DAO shouldn't be coupled with SQL